### PR TITLE
Test suite refactoring

### PR DESCRIPTION
**Pull request by bigspider**
Changes in this PR:

- Launching speculos as a fixture for each test, rather than a session-level process that is manually ran
- Added decorator to allow choosing the automation file for each test
- Added prepare_tests.sh and clean_tests.sh scripts to simplify the test suite setup

This significantly increases the running time if the test suite as speculos is now restarted for each test, but I believe it's a price worth paying for more reliable unit tests (guaranteeing they do not depend on each other), and the additional customization power (choosing different automation files is crucial to properly test the UX).

Also added a few new tests, and fixed a tiny bug in the handler of INS_HASH_SIGN, that was failing to properly check that P1 or P2 are set to 0.

### CODE REVIEW

File: tests/prepare_tests.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+cd ..
+make clean
+make DEBUG=1  # compile optionally with PRINTF
```
Comment (TamtamHero) [inline tests/prepare_tests.sh]: please add `-j` to make this fast :pray:

File: tests/prepare_tests.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+mv bin/ tests/bitcoin-bin
+make clean
+make DEBUG=1 COIN=bitcoin_testnet
```
Comment (TamtamHero) [inline tests/prepare_tests.sh]: here as well

**Review by TamtamHero — CHANGES_REQUESTED** [review]

File: tests/prepare_tests.sh (hunk 1 above)
Comment (bigspider) [reply tests/prepare_tests.sh]: Done in eb1f8041.

File: tests/prepare_tests.sh (hunk 2 above)
Comment (bigspider) [reply tests/prepare_tests.sh]: Done in eb1f8041.

**Pull request merged**